### DISCUSSION TITLE
Add support for more USB controllers/adapters

### DISCUSSION THREAD

**Discussion by gjimenezf** [cathery/sys-con#90] [Controller requests]
Hi, could you add support for these:
Nvidia shield controller:  vid="0955" pid="7214" buttoncount="18" axiscount="8"
Mayflash N64 USB adapter 2 ports:  Pid: 3013 vid: 0e8f

Or tell me what is needed to add support myself

**Comment by cualquiercosa327** [comment]
Hello,i am no a expert but i think i can help you. Maybe i put some mistakes,if it is the case please cathery could correct?
I have modded it for support my dualshock3 clone but i have could no test it because i have no installed yet the sdk (I have no many free time and i need reinstall the pc before could use it) without broke the code (the easiest would be change the vid and pid from the mine but it would stop working ds3 and ds4,i decided add a extra option as cathery did with both version of ds4).
The first part,which i have some doubts ,is they are a problem for add new devices (i am no sure if it is because they use the free memory the module as the part "the controllers table reached its max size" can use or why).
Anyway you can create a fork and begining replacing one with yours.
It depends of the controller you have ,but i would use the xbox360 one as base .

The files you should change :

config_global.ini
configFile.cpp
ControllerTypes.h
mainLoop.cpp
mainLoop.h
config_xbox360.ini

good luck!

**Comment by Sumea** [comment]
I am also interested in general additional controller support.
One main avenue is Playstation 3 and maybe PS4 Arcade sticks that generally are just generic Dinput pads in electronic level though. I am not sure how they are but MadCatz etc. sticks would be great support as most arcade style controllers come wired only anyway, and switch has quite a few fighting games and arcade ports.
